AsyncDisplayKit/Details/ASAbstractLayoutController.mm

```diff
@@ -25,32 +24,32 @@ - (instancetype)init
     return nil;
   }
   
-  _tuningParameters = std::vector<ASRangeTuningParameters>(ASLayoutRangeTypeCount);
-  _tuningParameters[ASLayoutRangeTypeVisible] = {
+  _tuningParameters = std::vector<std::vector<ASRangeTuningParameters>> (ASLayoutRangeModeCount, std::vector<ASRangeTuningParameters> (ASLayoutRangeTypeCount));
+  
+  _tuningParameters[ASLayoutRangeModeMinimum][ASLayoutRangeTypeVisible] = {
```

Comment:
Maybe it is time to remove the visible type from range parameters? This used to exist because the old range controller implementation had individual range handlers and it helped propagate the pattern, but now we can quite easily assuming that they are zero and make the appropriate call in the range controller. If you see any value to having them be explicit, feel free to keep them.